keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/DoubleTensor.java

```diff
@@ -159,6 +159,8 @@ static DoubleTensor placeHolder(int[] shape) {
 
     double standardDeviation();
 
+    boolean equalsWithEps( DoubleTensor other , double epsilon );
```

Comment:
Can I suggest we call this `isCloseTo`? That aligns with the method name both in Hamcrest and Python.

Reply:
I'd expect `isCloseTo` to be a distance mertic - where as this compares each dimension independently.

Reply:
That's an interesting way of looking at it... if you consider the "Manhattan" distance metric then this is the definition of "close" in that metric!

Reply:
Wouldn't Manhattan distance be the sum of differences - where this ensures that it's within a hypercube with sides of length `2*epsilon`...
Regardless the word `close` seems to imply a distance based metric - which isn't how the epsilon gets applied -and I'd prefer specific naming even if it diverges from other languages... However this is the syntax used in `junit` and `nd4j`

Reply:
The plot thickens: JUnit just has `assertEquals(expected, actual, delta)`. We probably wouldn't want to call our method `equals` though!

Reply:
it does - but the depreciation notice on `assertEquals(double,double)` calls it `epsilon` :)

I'm not to bothered, but wonder if it's worth getting other people's opinons

Reply:
I think your name is good, except I'd favour the slightly more long-hand `equalsWithinEpsilon`.

Reply:
I like the longer form name `equalsWithinEpsilon` but it looks like Numpy uses `isclose()`, which creates a stronger argument for `isCloseTo(...)`. I think `equalsWithinEpsilon(..)` is more like what we're doing, which is an element-wise compare as opposed to a multidimensional distance calculation.